app/views/unattended/provisioning_templates/host_init_config/host_init_config_default.erb

```diff
@@ -10,10 +10,27 @@ oses:
 -%>
 #!/bin/bash
 
-set -e
+<% built_https = foreman_url('built').start_with?('https') -%>
 
 # This script enrolls a host with Foreman.
 # Host: <%= @host.name %>
+echo "# Running [<%= @host.name %>] host initial configuration"
+
+trap 'exit_and_cancel_build' ERR
+
+<% if built_https -%>
+SSL_CA_CERT=$(mktemp)
+cat << EOF > $SSL_CA_CERT
+<%= foreman_server_ca_cert %>
+EOF
+<% end -%>
+
+exit_and_cancel_build() {
+  echo '<%= @host.name %> initial configuration failed'
+  curl --silent --show-error <%= '--cacert $SSL_CA_CERT' if built_https -%> -o /dev/null --noproxy \* '<%= foreman_url('built') %>'
```

Comment:
built implies it worked. Shouldn't we rather note that it failed?
```suggestion
  curl --silent --show-error <%= '--cacert $SSL_CA_CERT' if built_https -%> -o /dev/null --noproxy \* '<%= foreman_url('failed') %>'
```

Looking at kickstart can be useful:
https://github.com/theforeman/foreman/blob/d849076e8fc6a8c99ba60e935098a25de2b89c8a/app/views/unattended/provisioning_templates/provision/kickstart_default.erb#L346-L352

We actually support posting a log if you have it. Not suggesting that you immediately use it, but it's good to know the support is there.

Reply:
This would require to have template with 'failed' kind, what should be the content of the template? I thought that build can be only `pending` or nothing.

Reply:
`foreman_url` should really read `foreman_url_for_unattended_controller` so I believe it's this code:
https://github.com/theforeman/foreman/blob/69f74205c062dffc9e7c721a685ec9095fb7714b/app/controllers/unattended_controller.rb#L32-L43

So it does look like you can POST up to 65535 bytes with text. Does that help?

Reply:
Yes, thanks! PR updated